### DISCUSSION TITLE
Word Synced Lyrics

### DISCUSSION THREAD

**Discussion by Suya1671** [navidrome/navidrome#2920] [Ideas]
It would be neat to have word-synced lyrics rather than line-synced lyrics. This means that a whole line is shown but each word lights up over time. Not sure if this needs to be proposed to the subsonic API itself.

Some lrc files have the following format from what I've found:
```lrc
[line length]  <Word time> word <Time between words>   <Word time> OtherWord
```

Alternatively, this could be let up to the client to support. In Feishin right now, lyrics which are word-synced look like this:
![image](https://github.com/navidrome/navidrome/assets/45307955/33f05673-6191-4ac9-8d10-9256b8715f24)

**Comment by MasterZack69** [comment]
why is there not a single comment here and why hasn't elrc and ttml not been implemented yet

**Reply by metalheim** [reply]
The project is open source, you're free to fork navidrome and implement all features you want

**Reply by Suya1671** [reply]
Can we limit this discussion to actual ideas about how to go about this? (ngl I forgot I made this. It's been a while)

**Reply by DomHeadroom** [reply]
I would like to see this feature implemented too

**Comment by wilywyrm** [comment]
This is an old discussion, but I wanted to restart it because subsonic has added a [v2 version to their lyrics spec which supports word-level timings](https://opensubsonic.netlify.app/docs/endpoints/getlyricsbysongid/), and [lrclib/lrcget has introduced their Lyricsfile yaml format](https://github.com/tranxuanthang/lrcget/blob/main/LYRICSFILE_CONCEPT.md) which includes word-level timings.

**Reply by Suya1671** [reply]
Well then, looks like this is now officially a navidrome issue. Uh, should this be kept as a discussion or moved into an issue?

**Reply by wilywyrm** [reply]
Eh, I guess it's part of the contribution guidelines. [Done](https://github.com/navidrome/navidrome/issues/5548). [PR is out.](https://github.com/navidrome/navidrome/pull/5549)

**Reply by wilywyrm** [reply]
Oof, well good news for you is that there's already a PR out for this from February: https://github.com/navidrome/navidrome/pull/5076

**Comment by Suya1671** [comment]
#5076 has been merged and will be released Soon:tm: I assume, thus I shall close this discussion. Thanks to the devs for this!
